File: examples/headfirst/15_Bridge/Erl/bridgePattern.erl
-module(bridgePattern).
-export([main/0]).

main() ->
	Vehicle1 = car:new(produce:new(), assemble:new()),
	Vehicle1 ! {self(), manufacture},
	
	Vehicle2 = bike:new(produce:new(), assemble:new()),
	Vehicle2 ! {self(), manufacture}.

File: examples/headfirst/15_Bridge/Java/BridgePattern.java
//Este fonte foi retirado de: http://javapapers.com/design-patterns/bridge-design-pattern/
//Ultimo acesso em Agosto de 2013

public class BridgePattern {
	 
	  public static void main(String[] args) {
	 
	    Vehicle vehicle1 = new Car(new Produce(), new Assemble());
	    vehicle1.manufacture();
	    Vehicle vehicle2 = new Bike(new Produce(), new Assemble());
	    vehicle2.manufacture();
	  }
}
